src/InstrumentationEngine/ProfilerManager.cpp

```diff
@@ -509,43 +509,50 @@ DWORD WINAPI CProfilerManager::ParseAttachConfigurationThreadProc(
                 CComBSTR bstrInstrumentationMethodNodeName;
                 IfFailRet(pInstrumentationMethodNode->GetName(&bstrInstrumentationMethodNodeName));
 
-                if (wcscmp(bstrInstrumentationMethodNodeName, _T("AddInstrumentationMethod")) == 0)
+                if (wcscmp(bstrInstrumentationMethodNodeName, _T("InstrumentationMethod")) == 0)
                 {
-                    CComBSTR bstrConfigPath;
-                    IfFailRet(pInstrumentationMethodNode->GetAttribute(_T("ConfigPath"), &bstrConfigPath));
-                    IfFalseRet(bstrConfigPath.Length() != 0, E_FAIL);
+                    CComBSTR bstrInstrumentationMethodType;
+                    IfFailRet(pInstrumentationMethodNode->GetAttribute(_T("xsi:type"), &bstrInstrumentationMethodType));
```

Comment:
>xsi:type [](start = 75, length = 8)

Since the schema from my C# code has AddInstrumentationMethod inherit from InstrumentationMethod, it will generate the following xml:

```
<InstrumentationMethods>
  <InstrumentationMethod xsi:type="AddInstrumentationMethod" ...
```

Reply:
I'll add some comments to EngineConfiguration.cs on how to avoid this.

---
In reply to: [387832811](https://github.com/microsoft/clrinstrumentationengine/pull/220#discussion_r387832811) [](ancestors = 387832811)

Reply:
Thanks for the tips!

---
In reply to: [387842071](https://github.com/microsoft/CLRInstrumentationEngine/pull/220#discussion_r387842071) [](ancestors = 387842071,387832811)